### DISCUSSION TITLE
Debugging why articles are being marked as read

### DISCUSSION THREAD

**Discussion by tracteurblinde** [FreshRSS/FreshRSS#6621] [Q&A] [answered]
I have found myself in a place where articles I haven't read are getting marked as read. This has been ongoing for a couple months, but as an example: This weekend I did not load my instance, but still found feeds that updated over the weekend with some articles already marked as read.

I am very willing to believe this is something I did, but I haven't really determined the rhyme and reason so I'd love some help trying to figure this out.

- Doesn't seem to be a feed problem:
  - I went through looking for duplicate guids/titles and inexperienced eyes didn't see anything fishy.
  - [Ars Technica](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Farstechnica.com%2Ffeed%2F) and [Washington Smoke Information](https://validator.w3.org/feed/check.cgi?url=https%3A%2F%2Fwww.blogger.com%2Ffeeds%2F7325117526675950090%2Fposts%2Fdefault) are two example feeds that were completely read despite having new articles between Friday and Monday.
  - However, this does seem to be affecting the majority of my feeds
- Doesn't seem to be a feed settings problem:
  - I don't have filters set, all the read on X settings are set to default.
  - ![Feed settings showing all defaults](https://github.com/user-attachments/assets/98d984d1-a238-485a-946b-8bdd524770d3)
 - Doesn't seem to be a global settings problem:
   - I have the global archiving/read settings all disabled except for on viewing article.
   - ![User reading settings showing only "jump to next unread sibling (feed or category)" and "when the article is viewed" checked](https://github.com/user-attachments/assets/14b0df26-cb2e-4b3c-97b0-4ba62fc77cb2)
- I've gone through and disabled the API and rotated my password in case this was a rogue client.

What would be the next steps to try to debug why these articles are being marked as read?

Thank you.

**Comment by Alkarex** [comment]
Hello,
Some common culprits are described on https://github.com/FreshRSS/FreshRSS/discussions/6101
Try to see if you can do some more debugging based on that. Findings welcome

**Reply by tracteurblinde** [reply]
Hello,

Thank you for your work and taking the time to respond.

I guess I thought I had gone through that thread which is why I'd formatted the debugging I've done so far the way I did. Was there something in particular you think I missed?

My next steps were going to be writing an extension that just dumps `entry_auto_read` events and see if that got me anywhere (and then hacking up `‎FreshRSS_Entry._isRead‎` if that didn't). Is there a different next step you'd recommend first?

Thanks again.

**Reply by Alkarex** [reply]
Ah, sorry, I thought you were complaining about articles being marked as "unread", but it is the opposite...
Next time you experience the issue, could you please post a screenshot?
Please provide some information about your system as well (versions, DB, etc.)

**Reply by Alkarex** [reply]
And by the way, I have a new patch to fix the "max unread" policy, with some refactoring. If you can, testing this branch would be welcome https://github.com/FreshRSS/FreshRSS/pull/6632

**Reply by Alkarex** [reply]
I cannot think of what the cause of your symptoms could be. It would indeed require putting some debug lines at various places

**Reply by tracteurblinde** [reply, marked as answer]
tldr: Issue is resolved.

Got time to get back to this. I was a bit naughty and changed two variables:
- postgres image from `15` -> `16-alpine`
- freshrss image from `1.24.1-alpine` -> `edge-alpine` to pick up #6632 

Between the 1.24.2-dev version of freshrss and the postgres upgrade (which included a backup/restore), articles are no longer being marked read (except where intended).

Thank you for your help and your contributions.
